sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/algorithms/PIC.scala

```diff
@@ -1,14 +1,80 @@
 package net.sansa_stack.ml.spark.clustering.algorithms
 
+import com.typesafe.config.ConfigFactory
+import org.apache.jena.graph.{ NodeFactory, Triple }
 import org.apache.spark.graphx.Edge
 import org.apache.spark.graphx.Graph
 import org.apache.spark.mllib.clustering.PowerIterationClustering
-import org.apache.spark.rdd._
-import org.apache.spark.sql._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+import scala.collection.mutable.{ ArrayBuffer, HashMap }
 
+import net.sansa_stack.ml.spark.clustering.datatypes.Distance
+import net.sansa_stack.ml.spark.clustering.utils._
 
-class PIC {
+class PIC(input: RDD[Triple]) extends Serializable with ClusterAlgo {
+  val prefixID = "<http://clustering.slipo.eu/poi/"
+  val prefixCategory = "<http://clustering.slipo.eu/hasCategories>"
+  val prefixCoordinate = "<http://clustering.slipo.eu/hasCoordinate>"
 
+  var noofcluster = 0
+  var noOfIter = 0
+  var oneHotClusters = Map[Int, Array[Long]]()
+  val spark = SparkSession.builder.getOrCreate()
+  import spark.implicits._
+
+  val conf = ConfigFactory.load()
+
+  val dataProcessing = new DataProcessing(spark, conf, input)
+  val pois = dataProcessing.pois
+
+  def setK(K: Int): this.type = {
+    noofcluster = K
+    println(noofcluster)
+    this
+  }
+  /**
+   * set maximum iterations for Kmeans,PIC etc
+   * @param iter
+   */
+  def setMaxIterations(iter: Int): this.type = {
+    noOfIter = iter
+    println(noOfIter)
+    this
+  }
+
+  def run(): RDD[(Int, List[Triple])] = {
+    val spark = SparkSession.builder.getOrCreate()
+    import spark.implicits._
+
+    val poiCategorySetVienna = pois.map(poi => (poi.poi_id, poi.categories.categories.toSet)).persist()
+    val poiCartesian = poiCategorySetVienna.cartesian(poiCategorySetVienna)
+    println("Cartesian: " + poiCartesian.count())
```

Comment:
Remove all the `println` methods from here.

Reply:
done!